### PR TITLE
MMR API cleanups

### PR DESCRIPTION
**Pull request by str4d**
Addresses the remaining comments from #118.

### CODE REVIEW

File: librustzcash/src/rustzcash.rs [hunk 1]
```diff
@@ -1363,8 +1349,8 @@ pub extern "system" fn librustzcash_mmr_hash_node(
     };
 
     unsafe {
-        slice::from_raw_parts_mut(h_ret, 32).copy_from_slice(&node.hash()[..]);
+        (*h_ret).copy_from_slice(&node.hash()[..]);
```
Comment (NikVolf) [inline librustzcash/src/rustzcash.rs]: could be just `*h_ret = node.hash()`

File: librustzcash/src/rustzcash.rs [hunk 2]
```diff
@@ -1332,7 +1318,7 @@ pub extern "system" fn librustzcash_mmr_delete(
     };
 
     unsafe {
-        slice::from_raw_parts_mut(rt_ret, 32).copy_from_slice(
+        (*rt_ret).copy_from_slice(
```
Comment (NikVolf) [inline librustzcash/src/rustzcash.rs]: could be just
```
        *rt_ret = tree
                .root_node()
                .expect("Just generated without errors, root should be resolving")
                .data()
                .subtree_commitment;
```

File: librustzcash/src/rustzcash.rs [hunk 3]
```diff
@@ -1242,8 +1228,8 @@ pub extern "system" fn librustzcash_mmr_append(
     p_len: size_t,
     // New node pointer
     nn_ptr: *const [u8; zcash_mmr::MAX_NODE_DATA_SIZE],
-    // Return of root commitment (32 byte hash)
-    rt_ret: *mut u8,
+    // Return of root commitment
+    rt_ret: *mut [u8; 32],
     // Return buffer for appended leaves, should be pre-allocated of log2(t_len)+1 length
```
Comment (daira) [inline librustzcash/src/rustzcash.rs]: (From review of #118 with @str4d.)

This is not well-defined when `t_len` is not a power of 2. Should it be `ceiling(log2(t_len)) + 1`, or is the `+1` intended to implement the ceiling?

File: librustzcash/src/tests/mmr.rs [hunk 4]
```diff
@@ -10,7 +10,7 @@ struct TreeView {
     extra: Vec<(u32, Entry)>,
 }
 
-fn draft(into: &mut Vec<(u32, Entry)>, vec: &Vec<NodeData>, peak_pos: usize, h: u32) {
+fn draft(into: &mut Vec<(u32, Entry)>, vec: &[NodeData], peak_pos: usize, h: u32) {
```
Comment (daira) [inline librustzcash/src/tests/mmr.rs]: Rename the second parameter to something that doesn't imply it's a `Vec`.

File: librustzcash/src/tests/mmr.rs [hunk 5]
```diff
@@ -85,8 +85,8 @@ fn prepare_tree(vec: &Vec<NodeData>) -> TreeView {
     }
 }
 
-fn preload_tree_append(vec: &Vec<NodeData>) -> (Vec<u32>, Vec<[u8; zcash_mmr::MAX_ENTRY_SIZE]>) {
-    assert!(vec.len() > 0);
+fn preload_tree_append(vec: &[NodeData]) -> (Vec<u32>, Vec<[u8; zcash_mmr::MAX_ENTRY_SIZE]>) {
```
Comment (daira) [inline librustzcash/src/tests/mmr.rs]: Rename the parameter to something that doesn't imply it's a `Vec`.

File: librustzcash/src/tests/mmr.rs [hunk 6]
```diff
@@ -107,9 +107,9 @@ fn preload_tree_append(vec: &Vec<NodeData>) -> (Vec<u32>, Vec<[u8; zcash_mmr::MA
 
 // also returns number of peaks
 fn preload_tree_delete(
-    vec: &Vec<NodeData>,
+    vec: &[NodeData],
```
Comment (daira) [inline librustzcash/src/tests/mmr.rs]: Rename the parameter to something that doesn't imply it's a `Vec`.

File: librustzcash/src/rustzcash.rs (hunk 3 above)
Comment (NikVolf) [reply librustzcash/src/rustzcash.rs]: why binary log2 of `t_len` is not well-defined? +1 is for ceiling, yes
I does really need to specify upper bound only, so maybe mention it there

**Comment by str4d** [comment]
Huh, I was sure I ran the tests locally before pushing...

File: librustzcash/src/rustzcash.rs (hunk 3 above)
Comment (str4d) [reply librustzcash/src/rustzcash.rs]: I changed this to `ceiling(log2(t_len))` to be unambiguous.

**Comment by str4d** [comment]
Comments addressed!

**Pull request merged**